src/socket.rs

```diff
@@ -0,0 +1,37 @@
+use std::convert::TryInto;
+use std::io;
+
+use winapi::shared::minwindef::INT;
+use winapi::um::{winnt::SHORT, winsock2};
+
+pub const POLLRDNORM: SHORT = winsock2::POLLRDNORM;
+pub const POLLRDBAND: SHORT = winsock2::POLLRDBAND;
+pub const POLLIN: SHORT = winsock2::POLLIN;
+pub const POLLPRI: SHORT = winsock2::POLLPRI;
+pub const POLLWRNORM: SHORT = winsock2::POLLWRNORM;
+pub const POLLOUT: SHORT = winsock2::POLLOUT;
+pub const POLLWRBAND: SHORT = winsock2::POLLWRBAND;
+pub const POLLERR: SHORT = winsock2::POLLERR;
+pub const POLLHUP: SHORT = winsock2::POLLHUP;
+pub const POLLNVAL: SHORT = winsock2::POLLNVAL;
+
+pub use winsock2::WSAPOLLFD;
+
+/// `wsa_poll` waits for one of a set of file descriptors to become ready to perform I/O.
```

Comment:
Please keep a style consistent with the surrounding code. Namely, please wrap all code and comments to 79 columns, inclusive.